client/asset/btc/btc.go

```diff
@@ -1257,14 +1264,15 @@ func (btc *ExchangeWallet) sendWithReturn(baseTx *wire.MsgTx, addr btcutil.Addre
 					totalIn, totalOut, reqFee, fee)
 				return nil, nil, fmt.Errorf("change error")
 			}
-			if fee == reqFee || (reqFee < fee && tried[reqFee] == 1) {
+			if fee == reqFee || (fee > reqFee && tried[reqFee]) {
 				// If a lower fee appears available, but it's already been attempted and
 				// had a longer serialized size, the current fee is likely as good as
 				// it gets.
 				break
 			}
-			// We must have some room for improvement
-			tried[fee] = 1
+			// We must have some room for improvement.
+			sigCycles++
```

Comment:
Moving `sigCycles++` here won't count the initial cycle of the `for` loop. I was counting the original `SignTx` from the beginning of the function (with no change output) too.

Reply:
Fair enough.  Although I don't think it's possible to loop more than once now.   `reqFee` is going to be equal to `fee` on the first iteration unless SignTx change the serialized size.

Reply:
> `reqFee` is going to be equal to fee on the first iteration unless `SignTx` change the serialized size.

That is the hope. The loop is just to account for the case when the serialized size does change with new signatures.

Reply:
Understood, it's safest this way.  Although considering that the inputs don't change, it's not clear how the signatures would change.  Perhaps I'm missing a detail of txn signing beyond the input signature scripts that could change the serialized size.

Reply:
Oh, nvm there's a signature of a stripped version of the whole txn minus inputs.  Forget it! :)